mlrun/serving/states.py

```diff
@@ -981,25 +981,32 @@ def __init__(
         model_selector: Optional[Union[str, ModelSelector]] = None,
         **kwargs,
     ):
-        self._models = []
         super().__init__(
             *args,
             class_name="mlrun.serving.ModelRunner",
-            class_args=dict(runnables=self._models, model_selector=model_selector),
+            class_args=dict(model_selector=model_selector),
             **kwargs,
         )
 
-    def add_model(self, model: Model) -> None:
+    def add_model(self, model: Union[str, Model], **model_parameters) -> None:
         """Add a Model to this ModelRunner."""
-        self._models.append(model)
+        models = self.class_args.get("models", [])
+        models.append((model, model_parameters))
+        self.class_args["models"] = models
```

Comment:
not sure you need this line... list is mutable

Reply:
Yup, you need it in case the list didn't exist already.